shell/generate-deployment-command.sh

```diff
@@ -0,0 +1,181 @@
+    chains["bnb_smart_chain"]="$BSC_RPC_URL $BSCSCAN_API_KEY $BSC_CHAIN_ID $BSC_ADMIN $BSC_COMPTROLLER"
+    chains["gnosis"]="$GNOSIS_RPC_URL $GNOSISSCAN_API_KEY $GNOSIS_CHAIN_ID $GNOSIS_ADMIN $GNOSIS_COMPTROLLER"
+    chains["mainnet"]="$MAINNET_RPC_URL $ETHERSCAN_API_KEY $MAINNET_CHAIN_ID $MAINNET_ADMIN $MAINNET_COMPTROLLER"
+    chains["optimism"]="$OPTIMISM_RPC_URL $OPTIMISTIC_API_KEY $OPTIMISM_CHAIN_ID $OPTIMISM_ADMIN $OPTIMISM_COMPTROLLER"
+    chains["polygon"]="$POLYGON_RPC_URL $POLYGONSCAN_API_KEY $POLYGON_CHAIN_ID $POLYGON_ADMIN $POLYGON_COMPTROLLER"
+    chains["sepolia"]="$SEPOLIA_RPC_URL $ETHERSCAN_API_KEY $SEPOLIA_CHAIN_ID $SEPOLIA_ADMIN $SEPOLIA_COMPTROLLER"
+    chains["scroll"]="$SCROLL_RPC_URL $SCROLLSCAN_API_KEY $SCROLL_CHAIN_ID $SCROLL_ADMIN $SCROLL_COMPTROLLER"
+else
+    # Set bool to enter intaractive mode
+    INTERACTIVE=true
+
+    # Define the chain configurations
+    chains["arbitrum"]="$ARBITRUM_CHAIN_ID $ARBITRUM_COMPTROLLER"
+    chains["arbitrum_sepolia"]="$ARBITRUM_SEPOLIA_CHAIN_ID $ARBITRUM_SEPOLIA_COMPTROLLER"
+    chains["avalanche"]="$AVALANCHE_CHAIN_ID $AVALANCHE_COMPTROLLER"
+    chains["base"]="$BASE_CHAIN_ID $BASE_COMPTROLLER"
+    chains["bnb_smart_chain"]="$BSC_CHAIN_ID $BSC_COMPTROLLER"
+    chains["gnosis"]="$GNOSIS_CHAIN_ID $GNOSIS_COMPTROLLER"
+    chains["mainnet"]="$MAINNET_CHAIN_ID $MAINNET_COMPTROLLER"
+    chains["optimism"]="$OPTIMISM_CHAIN_ID $OPTIMISM_COMPTROLLER"
+    chains["polygon"]="$POLYGON_CHAIN_ID $POLYGON_COMPTROLLER"
+    chains["sepolia"]="$SEPOLIA_CHAIN_ID $SEPOLIA_COMPTROLLER"
+    chains["scroll"]="$SCROLL_CHAIN_ID $SCROLL_COMPTROLLER"
+fi
+
+# Check if the provided chain is defined
+if [[ ! -v "chains[$arg]" ]]; then
+    echo -e "\n${EC}Invalid chain: Chain configuration for '$arg' not found.${NC}"
+    exit 1
+fi
+
+# Check for arguments passed to the script
+if [[ $INTERACTIVE == true ]]; then
+    # load values from the terminal prompt
+    echo -e "${WC}.env.deployment missing, entering interactive mode.... ${NC}\n"
+    echo -e "1. Enter RPC URL: \c"
+    read rpc_url
+
+    echo -e "2. Enter etherscan API key: \c"
+    read api_key
+
+    echo -e "3. Enter admin address: \c"
+    read admin
+
+    echo -e "4. Enter max segment: \c"
+    read MAX_SEGMENT_COUNT
+
+    echo -e "5. Using deterministic script? (yes/no): \c"
+    read IS_DETERMINISTIC
+
+    # Get the values from the chains array
+    IFS=' ' read -r chain_id comptroller <<< "${chains[$arg]}"
+else
+    # load values from the terminal prompt
+    echo -e "Using deterministic script? (yes/no): \c"
+    read IS_DETERMINISTIC
+
+    # Get the values from the chains array
+    IFS=' ' read -r rpc_url api_key chain_id admin comptroller <<< "${chains[$arg]}"
+fi
+
+# Convert to lower case
+IS_DETERMINISTIC=${IS_DETERMINISTIC,,}
+
+if [[ $IS_DETERMINISTIC == "yes" || $IS_DETERMINISTIC == "y" ]]; then
+    # Construct the command
+    deployment_command="forge script script/DeployDeterministicCore3.s.sol"
```

Comment:
Can we make it more universal? Can we pass the script we want to run? 

e.g. `./shell/generate-deployment-command.sh Core3 chain` or `./shell/generate-deployment-command.sh Core2 chain`

what do you say, would it be useful?

Reply:
Instead of that, how about it gives options to select the script (by scanning the `/script` folder) through interactive mode in terminal. Based on what you select, it generates the command output.

Reply:
It would be even better

Reply:
```bash
~ ./shell/deploy-multi-chain.sh --script
Please select a script:
1) script/Base.s.sol				   10) script/DeployDeterministicLockupDynamic.s.sol
2) script/DeployComptroller.s.sol		   11) script/DeployDeterministicLockupLinear.s.sol
3) script/DeployCore.s.sol			   12) script/DeployDeterministicNFTDescriptor.s.sol
4) script/DeployCore2.s.sol			   13) script/DeployLockupDynamic.s.sol
5) script/DeployCore3.s.sol			   14) script/DeployLockupLinear.s.sol
6) script/DeployDeterministicComptroller.s.sol	   15) script/DeployNFTDescriptor.s.sol
7) script/DeployDeterministicCore.s.sol		   16) script/GenerateSVG.s.sol
8) script/DeployDeterministicCore2.s.sol	   17) script/Init.s.sol
9) script/DeployDeterministicCore3.s.sol
#? 4
+ You selected script/DeployCore2.s.sol
Compiling the contracts...

Deployment on sepolia started...
```